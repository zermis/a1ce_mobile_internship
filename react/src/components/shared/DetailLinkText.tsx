import React from 'react';
import { Text, StyleSheet, TouchableOpacity, Linking } from 'react-native';
import Colors from '../../constants/Colors';

interface Props {
    children: string;
    style?: object;
}

const DetailLinkText = (props: Props) => {
    return (
        <TouchableOpacity
            style={styles.unorderedList}
            onPress={() => void Linking.openURL(props.children)}
        >
            <Text style={styles.bulletPoint}>{'\u2022'}</Text>
            <Text style={styles.textLink}>{props.children}</Text>
        </TouchableOpacity>
    );
};

const styles = StyleSheet.create({
    unorderedList: {
        flexDirection: 'row'
    },
    bulletPoint: {
        color: Colors.textLink,
        fontSize: 16
    },
    textLink: {
        flex: 1,
        paddingLeft: 5,
        color: Colors.textLink,
        fontFamily: 'ibm-regular',
        fontSize: 14,
        textDecorationLine: 'underline'
    }
});

export default DetailLinkText;
