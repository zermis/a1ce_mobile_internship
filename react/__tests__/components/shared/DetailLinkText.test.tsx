import DetailLinkText from '../../../src/components/shared/DetailLinkText';
import renderer from 'react-test-renderer';

it('render detail link text correctly', () => {
    const tree = renderer
        .create(<DetailLinkText>{'Link is here'}</DetailLinkText>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
