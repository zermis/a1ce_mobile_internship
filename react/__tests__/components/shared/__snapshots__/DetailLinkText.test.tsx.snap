// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render detail link text correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#4884DE",
        "fontSize": 16,
      }
    }
  >
    â€¢
  </Text>
  <Text
    style={
      Object {
        "color": "#4884DE",
        "flex": 1,
        "fontFamily": "ibm-regular",
        "fontSize": 14,
        "paddingLeft": 5,
        "textDecorationLine": "underline",
      }
    }
  >
    Link is here
  </Text>
</View>
`;
